source/extensions/filters/http/ext_proc/ext_proc.h

```diff
@@ -108,12 +108,20 @@ class ExtProcLoggingInfo : public Envoy::StreamInfo::FilterState::Object {
     }
   }
 
+  void setHttpResponseCodeDetails(
+    const absl::optional<std::string>& http_response_code_details) {
```

Comment:
I don't follow the semantics here: if it has value, whether "" or not, we will assign it. 

when will we call with absl::nullopt?

Reply:
if the underlying http stream info hasn't been set yet, which is possible.

Reply:
Is it the same if just passing in std::string http_response_code_details.  And change the check below is if(!http_response_code_details.empty()) ...

Reply:
pls comment on the method for that special